MHVLib/Models/Generated/Categories/MHVGoalsApi.m

```diff
@@ -62,45 +62,37 @@ + (NSURLSessionTask*)createGoalsWithGoalsWrapper:(MHVGoalsWrapper*)goalsWrapper
             NSError* error = [NSError errorWithDomain:kMHVGoalsApiErrorDomain code:kMHVGoalsApiMissingParamErrorCode userInfo:userInfo];
             completion(nil, error);
         }
-        return nil;
     }
-
+    
     NSMutableString* resourcePath = [NSMutableString stringWithFormat:@"/Goals"];
-
+    
     NSMutableDictionary *pathParams = [[NSMutableDictionary alloc] init];
-
+    
     NSMutableDictionary* queryParams = [[NSMutableDictionary alloc] init];
-
+    
     id bodyParam = nil;
     NSMutableDictionary *formParams = [[NSMutableDictionary alloc] init];
     bodyParam = goalsWrapper;
-
-    return [MHVRemoteMonitoringClient requestWithPath: resourcePath
-                                    method:@"POST"
-                                pathParams:pathParams
-                               queryParams:queryParams
-                                formParams:formParams
-                                      body:bodyParam
-                                   toClass:[MHVSystemObject class]
-                                completion:^(id data, NSError *error)
-                                {
-                                    if(completion)
-                                    {
-                                        completion((MHVSystemObject*)data, error);
-                                    }
-                                }
-            ];
+    
+    [self requestWithPath:resourcePath
+               httpMethod:@"POST"
+               pathParams:pathParams
+              queryParams:queryParams
+               formParams:formParams
+                     body:bodyParam
```

Comment:
body is now NSData, which is what the iOS http service uses, so might need to add conversions here